storage/rocksdb/ha_rocksdb.h

```diff
@@ -91,6 +91,25 @@ struct Rdb_trx_info {
 std::vector<Rdb_trx_info> rdb_get_all_trx_info();
 
 /*
+ * class for exporting deadlock transaction information for
+ * information_schema.rocksdb_deadlock
+ */
+struct Rdb_deadlock_info {
+  struct Rdb_dl_trx_info {
+    ulonglong trx_id;
+    std::string cf_name;
+    std::string waiting_key;
+    bool exclusive_lock;
+    std::string index_name;
+    std::string table_name;
+  };
+  std::vector <Rdb_dl_trx_info> path;
+  ulonglong causer_trx_id;
```

Comment:
I think the intention of the message `TXN_ID GOT DEADLOCK` was to denote which transaction got rolled back. It's not so much the causer as it is the victim (although in rocksdb, the causer is *always* the victim).

How about a bool column called "Error" or "Rolled back" or something to denote that transaction got deadlock error? It would avoid repeating the id for every row.

Reply:
That actually makes a lot more sense than re-listing the transaction. Perhaps something similar can be added to the txn block output in the SHOW STATUS.